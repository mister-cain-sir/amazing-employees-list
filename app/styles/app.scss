@import "~bootstrap/scss/bootstrap-grid";
@import "~bootstrap/scss/mixins/_hover";
@import "~bootstrap/scss/vendor/rfs";
@import "~bootstrap/scss/mixins/_border-radius.scss";
@import "~bootstrap/scss/mixins/_table-row.scss";
@import "~bootstrap/scss/_navbar.scss";
@import "~bootstrap/scss/_tables";
