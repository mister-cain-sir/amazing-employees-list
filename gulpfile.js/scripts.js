const gulp = require("gulp"),
  webpack = require("webpack"),
  gulpWebpack = require("webpack-stream"),
  webpackConfig = require("../webpack.config");

class Scripts {
  constructor() {}
  buildLocal() {
    webpackConfig.mode = "development";
    return gulp
      .src("app/scripts/*.tsx")
      .pipe(gulpWebpack(webpackConfig, webpack))
      .pipe(gulp.dest("public/scripts"));
  }
  buildProd() {
    webpackConfig.mode = "production";
    return gulp
      .src("app/scripts/*.tsx")
      .pipe(gulpWebpack(webpackConfig, webpack))
      .pipe(gulp.dest("dist/scripts"));
  }
}
module.exports = Scripts;
