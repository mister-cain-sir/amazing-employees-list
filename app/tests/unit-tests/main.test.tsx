// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import {App} from "../../scripts/app";

test('App matches snapshot', () => {
  const component = renderer.create(
    <App></App>,
  );
  let app = component.toJSON();
  expect(app).toMatchSnapshot();
});