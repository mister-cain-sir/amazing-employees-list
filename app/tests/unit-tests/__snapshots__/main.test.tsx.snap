// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div
    className="fixed-top"
  >
    <div
      className="collapse"
      id="navbarToggleExternalContent"
    >
      <div
        className="bg-dark p-4"
      >
        <h5
          className="text-white h4"
        >
          Collapsed content
        </h5>
        <span
          className="text-muted"
        >
          Toggleable via the navbar brand.
        </span>
      </div>
    </div>
    <nav
      className="navbar navbar-dark bg-dark"
    >
      <button
        aria-controls="navbarToggleExternalContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarToggleExternalContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
    </nav>
  </div>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Age
        </th>
        <th>
          Date Of Birth
        </th>
        <th>
          Reporting Manager
        </th>
        <th>
          Department
        </th>
        <th>
          Salary
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          a
        </td>
        <td>
          23
        </td>
        <td>
          12/01/1987
        </td>
        <td>
          asdasd asfasfd
        </td>
        <td>
          aseasdaf
        </td>
        <td>
          234234234
        </td>
      </tr>
      <tr>
        <td>
          b
        </td>
        <td>
          25
        </td>
        <td>
          12/06/1977
        </td>
        <td>
          klhmhg toirhjrt
        </td>
        <td>
          oyptjns
        </td>
        <td>
          98876476585
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
